controllers/opentelemetrycollector_controller.go

```diff
@@ -164,6 +166,11 @@ func NewReconciler(p Params) *OpenTelemetryCollectorReconciler {
 				"ingresses",
 				true,
 			},
+			{
+				reconcile.ServiceMonitors,
+				"service monitors",
+				true,
+			},
```

Comment:
Will this task cause problems when the feature gate is `false`?

Reply:
The first thing to be checked during that task is if the feature flag is enabled:
```go
if !params.Instance.Spec.Observability.Metrics.CreateServiceMonitors || !featuregate.PrometheusOperatorIsAvailable.IsEnabled() {
```